Python/Python.sublime-syntax

```diff
@@ -421,20 +422,20 @@ contexts:
     # https://docs.python.org/3/reference/lexical_analysis.html#numeric-literals
     # hexadecimal
     - match: \b(?i)(0x)\h*(L) # py2
-      scope: constant.numeric.integer.long.hexadecimal.python
+      scope: constant.numeric.integer.hexadecimal.python
       captures:
         1: punctuation.definition.numeric.hexadecimal.python
-        2: storage.type.numeric.long.python
+        2: storage.type.numeric.python
     - match: \b(?i)(0x)(_?\h)+
       scope: constant.numeric.integer.hexadecimal.python
       captures:
         1: punctuation.definition.numeric.hexadecimal.python
     # octal
     - match: \b(?i)(0o?)(?=o|[0-7])[0-7]*(L) # py2
-      scope: constant.numeric.integer.long.octal.python
+      scope: constant.numeric.integer.octal.python
       captures:
         1: punctuation.definition.integer.octal.python
-        2: storage.type.numeric.long.python
```

Comment:
Why the removal of `long`?

Reply:
Because we are standardizing on `storage.type.numeric` for literal storage type suffixes across syntaxes. Certainly languages allow multiple suffixes, and by specializing the scope per letter suffix, we actually result in having multiple tokens for the suffix, which is odd, and messy. Plus, there isn't a whole lot of value in adding `.long` since you can infer that from `constant.numeric.integer storage.type.numeric.python`.

Reply:
> we actually result in having multiple tokens for the suffix,

can you elaborate on this?

Reply:
Ah, sorry - I think this comment was intended for line 434 - I can see the argument that having `long` in both inner and outer scopes is not useful.

Reply:
>> we actually result in having multiple tokens for the suffix,
> can you elaborate on this?

Every token in a text buffer in Sublime Text has a scope. Any time the scope changes, a there must, necessarily then, be a token boundary.

For example, imagine `123ri` in Ruby. The entire thing will have the scope `source.ruby constant.numeric.*.ruby`. However, the `ri` will also have `storage.type.numeric` since it is a suffix. Thus `123` is one token, and `ri` is another. If we specialize the suffix scope so that `r` gets `storage.type.numeric.rational.ruby` and `i` gets `storage.type.numeric.imaginary.ruby`, then we now have three tokens, `123`, `r` and `i`.

By **not** specializing the `r` and the `i`, we have two tokens, `123` and `ri`.